python/lsst/cp/verify/mergeResults.py

```diff
@@ -238,137 +365,233 @@ class CpVerifyRunMergeConnections(pipeBase.PipelineTaskConnections,
         dimensions=["instrument", "exposure"],
         multiple=True,
     )
+    inputResults = cT.Input(
+        name="exposureResults",
+        doc="Input results table to merge.",
+        storageClass="ArrowAstropy",
+        dimensions=["instrument", "exposure"],
+        multiple=True,
+    )
+    inputMatrix = cT.Input(
+        name="exposureMatrix",
+        doc="Input matrix table to merge.",
+        storageClass="ArrowAstropy",
+        dimensions=["instrument", "exposure"],
+        multiple=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        storageClass="Camera",
+        doc="Input camera.",
+        dimensions=["instrument", ],
+        isCalibration=True,
+    )
 
     outputStats = cT.Output(
         name="runStats",
         doc="Output statistics.",
         storageClass="StructuredDataDict",
-        dimensions=["instrument"],
+        dimensions=["instrument", ],
+    )
+    outputResults = cT.Output(
+        name="runResults",
+        doc="Output merged results table.",
+        storageClass="ArrowAstropy",
+        dimensions=["instrument",],
+    )
+    outputMatrix = cT.Output(
+        name="runMatrix",
+        doc="Output merged matrix table.",
+        storageClass="ArrowAstropy",
+        dimensions=["instrument",],
     )
 
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
 
-class CpVerifyRunMergeConfig(pipeBase.PipelineTaskConfig,
+        if not self.config.hasMatrixCatalog:
+            self.inputs.remove("inputMatrix")
+            self.outputs.remove("outputMatrix")
+
+
+class CpVerifyRunMergeConfig(CpVerifyExpMergeConfig,
                              pipelineConnections=CpVerifyRunMergeConnections):
     """Configuration paramters for exposure stats merging.
     """
-    runStatKeywords = pexConfig.DictField(
-        keytype=str,
-        itemtype=str,
-        doc="Dictionary of statistics to run on the set of exposure values. The key should be the test "
-        "name to record in the output, and the value should be the `lsst.afw.math` statistic name string.",
-        default={},
+    thisDimension = pexConfig.Field(
+        dtype=str,
+        doc="Dimension name for this input.",
+        default="exposure",
     )
 
 
-class CpVerifyRunMergeTask(pipeBase.PipelineTask):
+class CpVerifyRunMergeTask(CpVerifyExpMergeTask):
     """Merge statistics from detectors together.
     """
     ConfigClass = CpVerifyRunMergeConfig
     _DefaultName = 'cpVerifyRunMerge'
 
-    def runQuantum(self, butlerQC, inputRefs, outputRefs):
-        inputs = butlerQC.get(inputRefs)
+    pass
 
-        dimensions = [dict(exp.dataId.required) for exp in inputRefs.inputStats]
-        inputs['inputDims'] = dimensions
 
-        outputs = self.run(**inputs)
-        butlerQC.put(outputs, outputRefs)
+# End ExpMerge/RunMerge
+# Begin ExpMergeByFilter/RunMergeByFilter
```

Comment:
If we want these comments here, there should be a space between them.  Or a space, then the comments, then another space and the new class?

Reply:
I've made the comments stick either to the first or last line as appropriate, and put two spaces between, which seems to keep flake8 happy.  I added them after editing the wrong nearly-identical class multiple times.